client/main.lua

```diff
@@ -615,26 +658,41 @@ RegisterNetEvent('qbx_vehicleshop:client:swapVehicle', function(data)
     local dataTargetVehicle = sharedConfig.shops[shopName].showroomVehicles[data.targetVehicle]
     if dataTargetVehicle.vehicle == data.toVehicle then return end
 
-    local closestVehicle = lib.getClosestVehicle(dataTargetVehicle.coords.xyz, 5, false)
-    if not closestVehicle then return end
-
     if not IsModelInCdimage(data.toVehicle) then
         lib.print.error(('Failed to find model for "%s". Vehicle might not be streamed?'):format(data.toVehicle))
         return
     end
 
-    DeleteEntity(closestVehicle)
-    while DoesEntityExist(closestVehicle) do
-        Wait(50)
+    local points = lib.points.getAllPoints()
+    local vehPoint = nil
+    for i = 1, #points do
+        local point = points[i]
+        if point.coords == dataTargetVehicle.coords.xyz then
+            vehPoint = point
+            break
+        end
```

Comment:
Does this have a chance of being laggy if there are a lot of points for some reason? Can't we cache this information beforehand so we can immediately access the point that we want without looping through everything?

Reply:
It only gives the points in scope for the resource not the whole points table. so based on the current config that should be about 22 points.